ci-operator/step-registry/gcp/provision/buildhost/gcp-provision-buildhost-commands.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+CLUSTER_NAME="${NAMESPACE}-${JOB_NAME_HASH}"
+NETWORK=${NETWORK:-}
+IMAGE_ARGS=""
+if [[ -z "${IMAGE_FAMILY}" ]] && [[ ! -z "${IMAGE_NAME}" ]] ; then
+   IMAGE_ARGS="--image=${IMAGE_NAME}"
+fi
+
+if [[ ! -z "${IMAGE_FAMILY}" ]] && [[ -z "${IMAGE_NAME}" ]] ; then
+   IMAGE_ARGS="--image-family=${IMAGE_FAMILY}"
+fi
+
+if [[ -z "${IMAGE_ARGS}" ]]; then
+  echo "image info not correct"
+  exit 1 
+fi
+
+
+#####################################
+##############Initialize#############
+#####################################
+
+workdir=`mktemp -d`
+curl -L https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_amd64 -o /tmp/yq && chmod +x /tmp/yq
```

Comment:
I guess this ref was copied form `gcp-provision-bastionhost`, now that file was updated to stop downloading `yq` binary, I think here also need to be updated accordingly.

Reply:
this still needs to be here because the tests won't run without yq installed there 
I think we do still need yq, unless there's a new way yq is getting added

Reply:
You can change:
```
ref:
  as: gcp-provision-buildhost
  from_image:
    namespace: ocp
    name: "4.10"
    tag: upi-installer
```
to 
```
  from_image:
    namespace: ocp
    name: "4.12"
    tag: upi-installer
```

`yq-go` is already installed in the image, you can directly run `yq-go` instead of downloading it. More details refer to https://github.com/openshift/installer/pull/6008

Reply:
This is not resolved yet, right?

Reply:
@jianlinliu I don't think we should change the YQ download location for now. The way this works is to build a fedora:36 image with the source, and then YQ is installed. So, everyone who wants to use the buildhost functionality would need to put YQ into their base image. (We are not using the installer image).

Reply:
From [gcp-provision-buildhost-ref.yaml](https://github.com/openshift/release/pull/30733/files#diff-9b5fe3c2a110c5426ec96c0893eb2017ee09fecaef826d45645debabbac37d5d), I saw this:
```
  from_image:
    namespace: ocp
    name: "4.12"
    tag: upi-installer
```
Seem like you are using installer image, and the image already installed yq inside itself.